tests/foreman/maintain/test_backup.py

```diff
@@ -0,0 +1,566 @@
+    sat_maintain, setup_backup_tests, module_synced_repos, backup_type
+):
+    """Take a backup, ensure that '--split-pulp-tar' option works
+
+    :id: ddc3609d-642f-4161-b7a1-54f4aa069c08
+
+    :setup:
+        1. repo with sufficient size synced to the server
+
+    :steps:
+        1. create a backup using the split option
+        2. check that appropriate files are created
+        3. check that pulp_data.tar fits the split size
+
+    :expectedresult:
+        1. backup succeeds
+        2. expected files are present in the backup
+        3. size of the pulp_data.tar smaller than provided value
+    """
+    subdir = f'{BACKUP_DIR}backup-{gen_string("alpha")}'
+    set_size = 100
+    result = sat_maintain.cli.Backup.run_backup(
+        backup_dir=subdir,
+        backup_type=backup_type,
+        options={'assumeyes': True, 'split-pulp-tar': f'{set_size}k'},
+    )
+    logger.info(result.stdout)
+    assert result.status == 0
+    assert 'FAIL' not in result.stdout
+
+    # Check for expected files
+    backup_dir = re.findall(fr'{subdir}\/satellite-backup-.*-[0-5][0-9]', result.stdout)[0]
+    files = sat_maintain.execute(f'ls -a {backup_dir}').stdout.split('\n')
+    files = [i for i in files if not re.compile(r'^\.*$').search(i)]
+
+    expected_files = OFFLINE_BACKUP_FILES if backup_type == 'offline' else ONLINE_BACKUP_FILES
+    assert set(files).issuperset(expected_files + CONTENT_FILES), assert_msg
+
+    # Check the split works
+    result = sat_maintain.execute(f'du {backup_dir}/{CONTENT_FILES[0]}')
+    pulp_tar_size = int(result.stdout.split('\t')[0])
+    assert pulp_tar_size <= set_size
+
+
+@pytest.mark.parametrize('backup_type', ['online', 'offline'])
+def test_positive_backup_caspule_features(
+    sat_maintain, setup_backup_tests, module_synced_repos, backup_type
+):
+    """Take a backup with capsule features as dns, tftp, dhcp, openscap
+
+    :id: 7ebe8fe3-e5c3-454e-8e20-fad0a9d5b464
+
+    :steps:
+        1. create a backup
+        2. check that appropriate files are created
+
+    :expectedresult:
+        1. backup succeeds
+        2. expected files are present in the backup
+    """
+    subdir = f'{BACKUP_DIR}backup-{gen_string("alpha")}'
+    features = 'dns,tftp,dhcp,openscap'
+    result = sat_maintain.cli.Backup.run_backup(
+        backup_dir=subdir,
+        backup_type=backup_type,
+        options={'assumeyes': True, 'features': features},
+    )
+    logger.info(result.stdout)
+    assert result.status == 0
+    assert 'FAIL' not in result.stdout
+
+    # Check for expected files
+    backup_dir = re.findall(fr'{subdir}\/satellite-backup-.*-[0-5][0-9]', result.stdout)[0]
+    files = sat_maintain.execute(f'ls -a {backup_dir}').stdout.split('\n')
+    files = [i for i in files if not re.compile(r'^\.*$').search(i)]
+
+    expected_files = OFFLINE_BACKUP_FILES if backup_type == 'offline' else ONLINE_BACKUP_FILES
+    assert set(files).issuperset(expected_files + CONTENT_FILES), assert_msg
```

Comment:
Is it possible to check for features in backup files?

Reply:
We could probably check that the `config_files.tar.gz` contains the appropriate etc files. :thinking: